openpype/hosts/maya/plugins/publish/extract_camera_mayaScene.py

```diff
@@ -295,3 +301,43 @@ def transfer_image_planes(source_cameras, target_cameras,
 def _attach_image_plane(camera, image_plane):
     cmds.imagePlane(image_plane, edit=True, detach=True)
     cmds.imagePlane(image_plane, edit=True, camera=camera)
+
+
+@contextlib.contextmanager
+def lock_attributes(baked_camera_shapes, source_camera, lock_enabled):
+    # Need to include baked camera and source camera
```

Comment:
Why does it?

This is a `lib` function, this makes no sense to even be camera specific. 
This should just be:

```python
def locked_attributed(plugs, lock=True):
    original = {}
    try:
        for plug in plugs:
            original[plug] = cmds.getAttr(plug, lock=True)
            cmds.setAttr(plug, lock=lock)
        yield
    finally:
        for plug, state in original.items():
            cmds.setAttr(plug, lock=state)
```

It's then up to the caller of this function to use it the way they want.
I believe there's already a context manager dealing with attribute locking somewhere actually - used for camera extraction as well. There is e.g. this one: https://github.com/ynput/OpenPype/blob/c10781a662e9dd85839d8eb786e471b0709cbccf/openpype/hosts/maya/api/setdress.py#L79-L105

Also, be wary that if the original camera is e.g. a camera rig (or any reference) that it's likely that performing `setAttr` with the `lock` flag will error since you don't have full control over locking/unlocking of referenced attributes if e.g. the attribute is already locked in the reference.

So this implementation is pretty prone to error.